File: middle/src/types/Order.ts
export type OrderStatus =
  | 'pending_payment'
  | 'pending_shipment'
  | 'shipped'
  | 'completed'
  | 'closed'
  | 'after_sale';

export type PayStatus = 'unpaid' | 'paid' | 'refunded';

export type RefundStatus = 'none' | 'requested' | 'approved' | 'rejected' | 'refunded';

export interface OrderItem {
  productId: string;
  title: string;
  quantity: number;
  price: number;
}

export interface Order {
  id: string;
  orderNo: string;
  status: OrderStatus;
  payStatus: PayStatus;
  refundStatus: RefundStatus;
  totalAmount: number;
  itemCount: number;
  buyerName: string;
  shopId: string;
  shopName: string;
  createdAt: string;
  items: OrderItem[];
}

File: backend/src/main/java/io/cjc/backend/entity/Order.java
package io.cjc.backend.entity;

import io.cjc.backend.enums.OrderStatus;
import io.cjc.backend.enums.PayStatus;
import io.cjc.backend.enums.RefundStatus;
import jakarta.persistence.*;
import lombok.Data;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "orders")
@Data
public class Order {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private String id;

    @Column(name = "order_no", nullable = false, unique = true)
    private String orderNo;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private OrderStatus status = OrderStatus.PENDING_PAYMENT;

    @Enumerated(EnumType.STRING)
    @Column(name = "pay_status", nullable = false)
    private PayStatus payStatus = PayStatus.UNPAID;

    @Enumerated(EnumType.STRING)
    @Column(name = "refund_status", nullable = false)
    private RefundStatus refundStatus = RefundStatus.NONE;

    @Column(name = "total_amount", nullable = false, precision = 10, scale = 2)
    private BigDecimal totalAmount;

    @Column(name = "item_count", nullable = false)
    private Integer itemCount = 0;

    @Column(name = "buyer_id", nullable = false)
    private String buyerId;

    @Column(name = "buyer_name")
    private String buyerName;

    @Column(name = "shop_id", nullable = false)
    private String shopId;

    @Column(name = "shop_name")
    private String shopName;

    @Column(columnDefinition = "TEXT")
    private String remark;

    @Column(name = "receiver_name")
    private String receiverName;

    @Column(name = "receiver_phone")
    private String receiverPhone;

    private String province;

    private String city;

    private String district;

    @Column(name = "address_detail")
    private String addressDetail;

    @OneToMany(mappedBy = "order", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<OrderItem> items = new ArrayList<>();

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
}
